include/flox/resolver/descriptor.hh

```diff
@@ -0,0 +1,182 @@
+/* ========================================================================== *
+ *
+ * @file flox/resolver/descriptor.hh
+ *
+ * @brief A set of user inputs used to set input preferences and query
+ *        parameters during resolution.
+ *
+ *
+ * -------------------------------------------------------------------------- */
+
+#pragma once
+
+#include <string>
+#include <functional>
+#include <vector>
+
+#include <nlohmann/json.hpp>
+
+#include "flox/pkgdb/pkg-query.hh"
+#include "flox/resolver/params.hh"
+#include "flox/registry.hh"
+
+
+/* -------------------------------------------------------------------------- */
+
+namespace flox::resolver {
+
+/* -------------------------------------------------------------------------- */
+
+/**
+ * @brief Extend and remap fields from @a flox::resolver::PkgDescriptorRaw to
+ *        those found in a `flox` _manifest_.
+ *
+ * This _raw_ struct is defined to generate parsers.
+ * The _real_ form is @a flox::resolver::ManifestDescriptor.
+ */
+struct ManifestDescriptorRaw {
+
+  public:
+
+    /** Match `name`, `pname`, or `pkgAttrName` */
+    std::optional<std::string> name;
+
+    /** Match `version` or `semver` if a modifier is present. */
+    std::optional<std::string> version;
+
+    /** Match a catalog stability. */
+    std::optional<std::string> stability;
+
+    /** @brief A dot separated attribut path, or list representation. */
+    using Path = std::variant<std::string, flox::AttrPath>;
```

Comment:
Should we settle on a single format for these? Would this expose either format to users?

Reply:
I'd prefer a list, but the spec indicates that either a list of strings or a "dot separated path" may be used.

Reply:
I don't get the sense from the spec that's deliberate - if it is deliberate, great. If not, I think we should tweak it

Reply:
@graham made this tweak when I bumped into this a few weeks back so I think we're good.

Reply:
it was deliberate at the time but I wasn't thorough when I PRd it. it snuck in on another. if we could go back to the spec just calling for `.` then i'm good. if there's a "some people will prefer lists and it's near zero cost to implement both" then i'm good. even if we had to do lists it's not the end of the world because this way of expressing packages will likely be something advanced users do. 

I can put it back to dots since it seems like we now support both.